### PR TITLE
Use blake2b everywhere

### PR DESCRIPTION
**Pull request by jonas-lj**
## Description 

Closing #9225 

Previously, sui address is computed as the first 32 bytes of the SHA3-256 hash of `flag || pubkey`, this changes uses 
first 32 bytes of the Blake2bhash of `flag || pubkey` instead. This is for consideration of speed optimization. 

In Typescript, see https://github.com/MystenLabs/sui/blob/312c2be35eb902df8dfc90fbf8ca2e2d8ccd14ad/sdk/typescript/src/cryptography/ed25519-publickey.ts#L71

https://github.com/MystenLabs/sui/blob/312c2be35eb902df8dfc90fbf8ca2e2d8ccd14ad/sdk/typescript/src/cryptography/secp256k1-publickey.ts#L71

In Rust see: https://github.com/MystenLabs/sui/blob/312c2be35eb902df8dfc90fbf8ca2e2d8ccd14ad/crates/sui-types/src/base_types.rs#L466

## Test Plan 

All tests passes.

### Release notes
We now use the BLAKE2b-256 cryptographic hash function everywhere instead of Sha3-256. It is considered to be just as secure as Sha3 but is about three times faster. See also https://www.blake2.net.

### CODE REVIEW

**Comment by jonas-lj** [comment]
@pchrysochoidis could you update the wallet in this PR also (re #9248) to use Blake2b?

**Comment by pchrysochoidis** [comment]
> @pchrysochoidis could you update the wallet in this PR also (re #9248) to use Blake2b?

Just to repeat here [this comment](https://github.com/MystenLabs/sui/issues/9248#issuecomment-1467897500). Wallet does't do any hashing (as far as I know). It uses ts sdk that does the work for us. [here](https://github.com/MystenLabs/sui/blob/a0955c47974f73933edb832c664740502d19b55a/sdk/typescript/src/cryptography/ed25519-publickey.ts#L70), [here](https://github.com/MystenLabs/sui/blob/a0955c47974f73933edb832c664740502d19b55a/sdk/typescript/src/cryptography/secp256k1-publickey.ts#L70) and [here](https://github.com/MystenLabs/sui/blob/47a4ac3dd20889daeb723c38b15a6c8075276174/sdk/typescript/src/builder/TransactionData.ts#L123) are a few places I found but might be more. Ideally someone from devX should have a better idea.

**Review by kchalkias — COMMENTED** [review]
@joyqvq are we going to add the TS changes we explored (+ what @pchrysochoidis points out) in this PR?

**Review by kchalkias — COMMENTED** [review]
@pchrysochoidis all mentioned impact in the TS files you mentioned has been addressed. Please review

File: crates/sui-types/src/crypto.rs [hunk 1]
```diff
@@ -76,11 +76,7 @@ pub const DERIVATION_PATH_COIN_TYPE: u32 = 784;
 pub const DERVIATION_PATH_PURPOSE_ED25519: u32 = 44;
 pub const DERVIATION_PATH_PURPOSE_SECP256K1: u32 = 54;
 
-/// Default hash function for user-facing purposes, namely the ones that is also done by the wallet.
-pub type UserHash = Sha3_256;
-
-/// Default hash function for non-user-facing purposes, namely the ones not done by the wallet.
-pub type InternalHash = Blake2b256;
+pub type DefaultHash = Blake2b256;
```
Comment (kchalkias) [inline crates/sui-types/src/crypto.rs:79]: Add comment:
/// Blake2b_256 (output size = 32 bytes) is the default hash function for all Sui related operations including consensus and user-facing digests, such as publicKey to address and transactionID derivation.

**Review by kchalkias — COMMENTED** [review]
Let's rebase as well

**Review by joyqvq — APPROVED** [review]

**Review by kchalkias — COMMENTED** [review]
amazing work ethics here, thanks!

File: sdk/typescript/test/unit/cryptography/ed25519-publickey.test.ts [hunk 2]
```diff
@@ -83,13 +70,6 @@ describe('Ed25519PublicKey', () => {
     expect(new Ed25519PublicKey(key.toBytes()).equals(key)).toBe(true);
   });
 
-  it('toSuiAddress', () => {
```
Comment (Jordan-Mysten) [inline sdk/typescript/test/unit/cryptography/ed25519-publickey.test.ts:86]: Curious why this test was removed?

Comment (joyqvq) [reply sdk/typescript/test/unit/cryptography/ed25519-publickey.test.ts:86]: https://github.com/MystenLabs/sui/pull/9262/files/312c2be35eb902df8dfc90fbf8ca2e2d8ccd14ad#diff-c4c301721c06ce4af4e64ffa7d26b33ddff61631a71a574abcf532a74c583d79R73 

there are a bunch of tests here already for the same purpose

**Pull request merged**